extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+
+[tip type="note"]
+Calling `AMP.setState()` in some examples may set or change states of other examples on page. Refresh this page to see examples before `AMP.setState()`.
+[/tip]
 
 ### A slightly more complex example
 
 [filter formats="websites, stories, ads"]
 
+[example preview="top-frame" playground="true" imports="amp-bind"]
+
+```html
+<head>
+  <style amp-custom>
+    .greenBorder {
+      border: 5px solid green;
+    }
+    .redBorder {
+      border: 5px solid red;
+    }
+  </style>
+</head>
+<body>
+  <amp-state id="theFood">
+    <script type="application/json">
+      {
+        "cupcakes": {
+          "imageUrl": "https://amp.dev/static/samples/img/image2.jpg",
+          "style": "greenBorder"
+        },
+        "sushi": {
+          "imageUrl": "https://amp.dev/static/samples/img/image3.jpg",
+          "style": "redBorder"
+        }
+      }
+    </script>
+  </amp-state>
+  <div class="greenBorder" [class]="theFood[currentMeal].style">
+    <p>Each food has a different border color.</p>
+    <p [text]="'I want to eat ' + currentMeal + '.'">I want to eat cupcakes.</p>
+    <amp-img
+      width="300"
+      height="200"
+      src="https://amp.dev/static/samples/img/image2.jpg"
+      [src]="theFood[currentMeal].imageUrl"
+    >
+    </amp-img>
+    <button on="tap:AMP.setState({currentMeal: 'sushi'})">Set to sushi</button>
+    <button on="tap:AMP.setState({currentMeal: 'cupcakes'})">
+      Set to cupcakes
+    </button>
+  </div>
+</body>
+```
+
+[/example]
+
+In the example above:
+
+- The `<amp-state>` component declares state using a JSON string literal. It has an `id` of `theFood` to allow us to reference the defined data. But because `<amp-bind>` does not evaluate `<amp-state>` on page load, the **state** is empty.
+- The page loads with visual defaults.
+  - The `<div>` element has `class="greenBorder"` defined.
+  - The second `<p>` element has "I want cupcakes." defined within the tags.
+  - The `<amp-img>` `src` points to a url.
+- Changeable elements have **bindings** that point to **expressions**.
+  - The `[class]` attribute on the `<div>` is bound to the `theFood[currentMeal].style` **expression**.
+  - The `[text]` attribute on the second `<p>` is bound to the `'I want to eat ' + currentMeal + '.'` **expression**.
+  - The `[src]` attribute is bound to the `theFood[currentMeal].imageUrl` **expression**.
+
+If a user clicks the "Set to sushi" button:
+
+1. The `tap` event trigger the `AMP.setState` action.
+1. The setState action turns `currentMeal` into a state and sets it to `sushi`.
+1. AMP evaluates **bindings** with **expressions** that contain the state `currentMeal`.
+1. `[class]="theFood[currentMeal].style"` updates `class` to `redBorder`.
+1. `[text]="'I want to eat ' + currentMeal + '.'"` updates the inner text of the second `<p>` element to "I want to eat sushi".
+1. `[src]="theFood[currentMeal].imageUrl` updates the `src` of `<amp-img>` to `https://amp.dev/static/samples/img/image3.jpg`
```

Comment:
Here are these all #1 still or is this a shorthand for numerical items in a list in markdown?

Reply:
;)

Reply:
Its so you don't mess up when you count and can move/remove/add items with ease. yay markdown!